### PR TITLE
Documenting renderBlockingStatus in Resource Timing

### PR DESCRIPTION
**Pull request by abinpaul1**
#### Summary

New field `renderBlockingStatus` in PerfomanceResourceTiming to indicate the render-blocking nature of a resource. The attribute takes away the burden of having to rely on complex heurestics to determine which resources were actually render-blocking and provides a direct signal from the browser.

Explainer : https://github.com/abinpaul1/resource-timing/blob/render-blocking-status-explainer/Explainer/Render_Blocking_Status.md

Spec changes : https://github.com/w3c/resource-timing/pull/327

#### Motivation

Adding documentation for new attribute`renderBlockingStatus` introduced to Resource Timing. Adding sample code to demonstrate usage of the attribute.

#### Supporting details

CR Bug : https://bugs.chromium.org/p/chromium/issues/detail?id=1337256
Chrome status entry : https://chromestatus.com/feature/5166965277589504

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by bsmth** [comment]
Hi @abinpaul1, thanks for opening the PR. I'm just checking if you're planning to pick this one up again or how you'd like to proceed. Thanks a lot :)

**Comment by abinpaul1** [comment]
I was thinking I'd have to wait for Chrome 107 to ship which would be the first browser release to have this enabled. I'm happy to continue working on this right away. Please feel free to start a review for this and I can adjust it based on feedback or I'm also open to handing it over if thats preferred.

File: files/en-us/web/api/performanceresourcetiming/index.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ The interface also supports the following properties which are listed in the ord
   - : A `number` that is the size (in octets) received from the fetch (HTTP or cache) of the _message body_, after removing any applied content-codings.
 - {{domxref('PerformanceResourceTiming.serverTiming')}} {{ReadOnlyInline}}
   - : An array of {{domxref("PerformanceServerTiming")}} entries containing server timing metrics.
+- {{domxref('PerformanceResourceTiming.renderBlockingStatus')}} {{readonlyInline}}
```
Comment (bsmth) [inline files/en-us/web/api/performanceresourcetiming/index.md:74]: Is this supposed to be `ReadOnlyInline` ?

```suggestion
- {{domxref('PerformanceResourceTiming.renderBlockingStatus')}} {{ReadOnlyInline}}
```

File: files/en-us/web/api/performanceresourcetiming/renderBlockingStatus/index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+- `"blocking"`
+  - : The resource might potentially block rendering.
+- `"non-blocking"`
+  - : The resource does not block rendering.
+
+## Examples
+
+In the following example, we print out the {{domxref("PerformanceEntry.name","name")}} of
+resources which are potentially render-blocking.
+```js
```
Comment (bsmth) [inline files/en-us/web/api/performanceresourcetiming/renderBlockingStatus/index.md:30]: This will need a newline before the code block

**Comment by bsmth** [comment]
> I'd have to wait for Chrome 107 to ship

Thanks a lot, I just had a look through [the Chrome bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1337256) and it looks like it's expected to ship soon, is that right? Feel free to ping me to have another look! :)

File: files/en-us/web/api/performanceresourcetiming/renderBlockingStatus/index.md (hunk 2 above)
Comment (abinpaul1) [reply files/en-us/web/api/performanceresourcetiming/renderBlockingStatus/index.md:32]: Added

File: files/en-us/web/api/performanceresourcetiming/index.md (hunk 1 above)
Comment (abinpaul1) [reply files/en-us/web/api/performanceresourcetiming/index.md]: Yes. Updated. Thanks

**Comment by abinpaul1** [comment]
Yes, stable is expected to ship towards end of October. I am going ahead and marking the PR as ready for review.

**Comment by bsmth** [comment]
OK thanks a lot. There will be some downtime from the team during the next week or so; we will get to this afterwards. I'm happy to pick it up then.

**Comment by abinpaul1** [comment]
@bsmth I'm not able to get around an error when I add this new document for renderBlockingStatus. I am trying to keep it similar to how all the other fields in Resource Timing are added, but I get this error. 
```
MacroBrokenLinkError from domxref [line 112:7] 
Explanation: /en-US/docs/Web/API/PerformanceResourceTiming/renderBlockingStatus does not exist `
```
I feel I'm misssing something to get the links to work correctly

**Comment by bsmth** [comment]
> I feel I'm misssing something to get the links to work correctly

It could be a case-sensitivity thing on the parent directory for the document. I can take a look shortly

**Comment by bsmth** [comment]
@abinpaul1 this is fixed now in https://github.com/mdn/content/pull/19325/commits/ac79f03bc5458c9f16d93c3fafd0979136736105, the parent dir is case-sensitive

**Comment by abinpaul1** [comment]
Cool! Thanks

File: files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - Property
+  - Reference
+  - Web Performance
+browser-compat: api.PerformanceResourceTiming.renderBlockingStatus
+---
+
+{{APIRef("Resource Timing API")}}
+
+The **`renderBlockingStatus`** read-only property returns the render-blocking
+nature of the resource.
```
Comment (bsmth) [inline files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md:16]: Could we give one or two more sentences on the context, what's being blocked for rendering and why should someone use this?

File: files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Examples
+
+In the following example, we print out the {{domxref("PerformanceEntry.name","name")}} of
+resources which are potentially render-blocking.
+
+```js
+function print_renderBlocking_resources() {
+  const p = performance.getEntriesByType("resource");
+  for (let i=0; i < p.length; i++) {
```
Comment (bsmth) [inline files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md:35]: The style guide prefers `forEach()` or `for ... of` -> https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#loops_and_conditional_statements

Comment (abinpaul1) [reply files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md]: Done

File: files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md (hunk 3 above)
Comment (abinpaul1) [reply files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md]: I've added a couple. Please have a look

Comment (bsmth) [reply files/en-us/web/api/performanceresourcetiming/renderblockingstatus/index.md]: Thank you, I'll take a look shortly. By the way: how is the feature support at the moment? I would also like to try out the example you provided to test

**Comment by bsmth** [comment]
I tested in Chrome `108.0.5355.0` and works as expected. This should be on-hold until 107 stable release, but I think we can publish it then. We will need corresponding browser compat data for this first.

**Comment by abinpaul1** [comment]
I've opened a PR for adding the compat data : https://github.com/mdn/browser-compat-data/pull/18046

**Comment by bsmth** [comment]
> I've opened a PR for adding the compat data : [mdn/browser-compat-data#18046](https://github.com/mdn/browser-compat-data/pull/18046)

oh nice, you beat me to it 🙌🏻

**Comment by abinpaul1** [comment]
Hey @bsmth, Are we good to merge this?

**Comment by bsmth** [comment]
Hi @abinpaul1, I think the corresponding BCD PR has to be merged first. Once that's landed I think we're ready to go here.

**Comment by bsmth** [comment]
- [x] https://github.com/mdn/browser-compat-data/pull/18046

BCD for this is merged, this PR can be merged when this is deployed (likely today).

**Comment by Josh-Cena** [comment]
This should be mergeable now?

**Review by bsmth — APPROVED** [review]
Thanks @abinpaul1, going to merge shortly

**Pull request merged**